src/coreclr/jit/loopcloning.cpp

```diff
@@ -862,6 +863,14 @@ BasicBlock* LoopCloneContext::CondToStmtInBlock(Compiler*
             insertAfter = newBlk;
         }
 
+        // The block preceding the newly-generated conditions could have been a BBJ_ALWAYS.
+        // Make sure it jumps to the new block chain inserted after it.
+        if (firstInsertAfter->KindIs(BBJ_ALWAYS))
+        {
+            assert(!firstInsertAfter->IsLast());
+            firstInsertAfter->SetJumpDest(firstInsertAfter->Next());
+        }
```

Comment:
Does this handle the preds properly?

Reply:
I believe so. Here's my logic: In the loop above this if-statement, in the first iteration, we insert a new block (`newBlk`) after `insertAfter`, and then call `fgAddRefPred(newBlk, insertAfter)`. At the end of the loop, we set `insertAfter = newBlk`, such that we can keep chaining new blocks after the one we just inserted. While the loop handles the preds correctly, it does not account for the possibility of the initial block `insertAfter` points to being a `BBJ_ALWAYS`, and thus does not update its jump destination. I didn't want to add this logic of updating the jump destination into the loop since it would be necessary at most once, so I added this logic after the loop body.

(I noticed this edge case in the first place by adding an assert to `fgAddRefPred` such that if `blockPred` is a `BBJ_ALWAYS`, then `blockPred->HasJumpTo(block)` must be true already, so that I wouldn't forget to update its jump destination).